compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -464,6 +463,12 @@ public void visit(BLangRecordLiteral recordLiteral) {
     }
 
     private boolean isRecordLiteralCompatible(BRecordType bRecordType, BLangRecordLiteral recordLiteral) {
+        if (recordLiteral.getKeyValuePairs().isEmpty()) {
```

Comment:
Do we need this check here. AFAIK we do a similar check at the `checkMissingRequiredFields(..)` method..